can't push/pull behind proxy

**Describe the bug**
Git ui is not able to push/pull when behind corporate proxy

**To Reproduce**
Steps to reproduce the behavior:
1. Commit a change
2. Try to push it (press p)
or 
1. open gitui
2. Try to pull ( press f )


**Expected behavior**
Being able to push commits to remote when running behind porxy

**Screenshots**
When pulling following message comes up
![image](https://user-images.githubusercontent.com/20717654/142343095-095bba1b-e1d9-490e-abaa-7aeda3cb6d44.png)

but when git pull is run from same terminal it works

```
$ git pull
Already up to date.
```



**Context (please complete the following information):**
 - PopOS (21.10) [ Ubuntu 21.10 ]
 - gitui 0.18.0
 - rustc 1.56.1 (59eed8a2a 2021-11-01)


Proxy is set using environment variables http_proxy & https_proxy

```
22:24:34 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:24:34 [TRACE] (6) mio::poll: [/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-0.7.13/src/poll.rs:478] registering event source with poller: token=Token(0), interests=READABLE
22:24:34 [TRACE] (6) mio::poll: [/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-0.7.13/src/poll.rs:478] registering event source with poller: token=Token(1), interests=READABLE
22:24:34 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 16737014557684109584] (type: WorkingDir)
22:24:34 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 2753998338531215346] (type: Stage)
22:24:34 [TRACE] (3) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 2753998338531215346 (type: Stage)
22:24:34 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:34 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 16737014557684109584 (type: WorkingDir)
22:24:34 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:36 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('f'), modifiers: NONE }))
22:24:36 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:24:36 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 11962389017709865077] (type: WorkingDir)
22:24:36 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 17874425802020491930] (type: Stage)
22:24:36 [TRACE] (4) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 17874425802020491930 (type: Stage)
22:24:36 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 11962389017709865077 (type: WorkingDir)
22:24:36 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:36 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:39 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('k'), modifiers: NONE }))
22:24:39 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('o'), modifiers: NONE }))
22:24:39 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('p'), modifiers: NONE }))
22:24:39 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:24:39 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 4889631339021475959] (type: WorkingDir)
22:24:39 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 9810875698262969305] (type: Stage)
22:24:39 [TRACE] (3) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 9810875698262969305 (type: Stage)
22:24:39 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 4889631339021475959 (type: WorkingDir)
22:24:39 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:39 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:40 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Backspace, modifiers: NONE }))
22:24:40 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Backspace, modifiers: NONE }))
22:24:40 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('p'), modifiers: NONE }))
22:24:41 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('2'), modifiers: NONE }))
22:24:41 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('4'), modifiers: NONE }))
22:24:42 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('0'), modifiers: NONE }))
22:24:42 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('1'), modifiers: NONE }))
22:24:42 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('0'), modifiers: NONE }))
22:24:43 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('8'), modifiers: NONE }))
22:24:43 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('5'), modifiers: NONE }))
22:24:43 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Backspace, modifiers: NONE }))
22:24:44 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Backspace, modifiers: NONE }))
22:24:44 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('7'), modifiers: NONE }))
22:24:44 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('5'), modifiers: NONE }))
22:24:44 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:24:44 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 1014738629262155162] (type: WorkingDir)
22:24:44 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 13387760825023286158] (type: Stage)
22:24:44 [TRACE] (4) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 13387760825023286158 (type: Stage)
22:24:44 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 1014738629262155162 (type: WorkingDir)
22:24:44 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:44 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:45 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('@'), modifiers: NONE }))
22:24:45 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('g'), modifiers: NONE }))
22:24:45 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('m'), modifiers: NONE }))
22:24:45 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('a'), modifiers: NONE }))
22:24:46 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('i'), modifiers: NONE }))
22:24:46 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('l'), modifiers: NONE }))
22:24:46 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('.'), modifiers: NONE }))
22:24:46 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('c'), modifiers: NONE }))
22:24:46 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('o'), modifiers: NONE }))
22:24:46 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('m'), modifiers: NONE }))
22:24:47 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE }))
22:24:49 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:24:49 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 1305773434620817606] (type: WorkingDir)
22:24:49 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 7394116204606378236] (type: Stage)
22:24:49 [TRACE] (3) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 7394116204606378236 (type: Stage)
22:24:49 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 1305773434620817606 (type: WorkingDir)
22:24:49 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:49 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:52 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Resize(83, 67))
22:24:54 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:24:54 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 4165908251595713894] (type: WorkingDir)
22:24:54 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 1970964189430421176] (type: Stage)
22:24:54 [TRACE] (4) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 1970964189430421176 (type: Stage)
22:24:54 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 4165908251595713894 (type: WorkingDir)
22:24:54 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:54 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:59 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:24:59 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 10701491168694815144] (type: WorkingDir)
22:24:59 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 6865505702600828986] (type: Stage)
22:24:59 [TRACE] (3) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 6865505702600828986 (type: Stage)
22:24:59 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 10701491168694815144 (type: WorkingDir)
22:24:59 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:24:59 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('g'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('h'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('p'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('_'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('0'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('6'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('w'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('U'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('o'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('l'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('M'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('W'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('q'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('v'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('M'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('N'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('R'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('O'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('H'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('y'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('o'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('y'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('n'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('1'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('l'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('0'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('r'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('N'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('V'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('w'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('g'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('u'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('U'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('2'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('2'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('B'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('5'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('2'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('P'), modifiers: SHIFT }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('t'), modifiers: NONE }))
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE }))
22:25:02 [TRACE] (1) asyncgit::fetch: [asyncgit/src/fetch.rs:69] request
22:25:02 [INFO] [Debug]: Sending GET request to https://github.com/kp2401075/qframe.git/info/refs?service=git-upload-pack
22:25:02 [INFO] [Debug]: Connecting to remote github.com port 443
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Fetch)
22:25:02 [ERROR] fetch error: git error:failed to resolve address for github.com: Name does not resolve; class=Net (12)
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:02 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 11504950566956201133] (type: WorkingDir)
22:25:02 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 15639663759393566724] (type: Stage)
22:25:02 [TRACE] (4) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 15639663759393566724 (type: Stage)
22:25:02 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 11504950566956201133 (type: WorkingDir)
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Fetch)
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:02 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:04 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:04 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 1333511300604058936] (type: WorkingDir)
22:25:04 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 528334033776148253] (type: Stage)
22:25:04 [TRACE] (4) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 528334033776148253 (type: Stage)
22:25:04 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 1333511300604058936 (type: WorkingDir)
22:25:04 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:04 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:06 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE }))
22:25:09 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:09 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 8828791648320725906] (type: WorkingDir)
22:25:09 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 475532260167004461] (type: Stage)
22:25:09 [TRACE] (3) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 475532260167004461 (type: Stage)
22:25:09 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 8828791648320725906 (type: WorkingDir)
22:25:09 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:09 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:14 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('f'), modifiers: NONE }))
22:25:14 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:14 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 11493445326400361978] (type: WorkingDir)
22:25:14 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 14968885388592189007] (type: Stage)
22:25:14 [TRACE] (3) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 14968885388592189007 (type: Stage)
22:25:14 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 11493445326400361978 (type: WorkingDir)
22:25:14 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:14 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:14 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:14 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 882734245316001000] (type: WorkingDir)
22:25:14 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 1828324095550698375] (type: Stage)
22:25:14 [TRACE] (4) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 1828324095550698375 (type: Stage)
22:25:14 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 882734245316001000 (type: WorkingDir)
22:25:14 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:14 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:15 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE }))
22:25:18 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE }))
22:25:18 [TRACE] (1) asyncgit::fetch: [asyncgit/src/fetch.rs:69] request
22:25:18 [INFO] [Debug]: Sending GET request to https://github.com/kp2401075/qframe.git/info/refs?service=git-upload-pack
22:25:18 [INFO] [Debug]: Connecting to remote github.com port 443
22:25:18 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Fetch)
22:25:18 [ERROR] fetch error: git error:failed to resolve address for github.com: Name does not resolve; class=Net (12)
22:25:18 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:18 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 12793033651671889328] (type: WorkingDir)
22:25:18 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 7718095315411454769] (type: Stage)
22:25:18 [TRACE] (3) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 7718095315411454769 (type: Stage)
22:25:18 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 12793033651671889328 (type: WorkingDir)
22:25:18 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Fetch)
22:25:18 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:18 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:19 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE }))
22:25:19 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:19 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 10632165581298515111] (type: WorkingDir)
22:25:19 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 11615974367102171234] (type: Stage)
22:25:19 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 10632165581298515111 (type: WorkingDir)
22:25:19 [TRACE] (4) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 11615974367102171234 (type: Stage)
22:25:19 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:19 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:24 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:24 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 7346829671670689732] (type: WorkingDir)
22:25:24 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 16784878880735985319] (type: Stage)
22:25:24 [TRACE] (3) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 16784878880735985319 (type: Stage)
22:25:24 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 7346829671670689732 (type: WorkingDir)
22:25:24 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:24 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:24 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('f'), modifiers: NONE }))
22:25:24 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:24 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 16254706157462405849] (type: WorkingDir)
22:25:24 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 920060072598101741] (type: Stage)
22:25:24 [TRACE] (4) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 920060072598101741 (type: Stage)
22:25:24 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 16254706157462405849 (type: WorkingDir)
22:25:24 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:24 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:25 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('S'), modifiers: SHIFT }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('h'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('a'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('n'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('i'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('d'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('e'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('v'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('1'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('9'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('9'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('5'), modifiers: NONE }))
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE }))
22:25:26 [TRACE] (1) asyncgit::fetch: [asyncgit/src/fetch.rs:69] request
22:25:26 [INFO] [Debug]: Sending GET request to https://github.com/kp2401075/qframe.git/info/refs?service=git-upload-pack
22:25:26 [INFO] [Debug]: Connecting to remote github.com port 443
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Fetch)
22:25:26 [ERROR] fetch error: git error:failed to resolve address for github.com: Name does not resolve; class=Net (12)
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:26 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 13147674871717321165] (type: WorkingDir)
22:25:26 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 15642236361445625405] (type: Stage)
22:25:26 [TRACE] (3) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 15642236361445625405 (type: Stage)
22:25:26 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 13147674871717321165 (type: WorkingDir)
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Fetch)
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:26 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:28 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE }))
22:25:29 [TRACE] (1) gitui::app: [src/app.rs:358] update
22:25:29 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 5701710530350939715] (type: WorkingDir)
22:25:29 [TRACE] (1) asyncgit::status: [asyncgit/src/status.rs:95] request: [hash: 2845519784891164607] (type: Stage)
22:25:29 [TRACE] (4) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 2845519784891164607 (type: Stage)
22:25:29 [TRACE] (2) asyncgit::status: [asyncgit/src/status.rs:151] status fetched: 5701710530350939715 (type: WorkingDir)
22:25:29 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:29 [TRACE] (1) gitui::app: [src/app.rs:377] update_async: Git(Status)
22:25:31 [TRACE] (1) gitui::app: [src/app.rs:286] event: Input(Key(KeyEvent { code: Char('c'), modifiers: CONTROL }))


```

--- a/asyncgit/src/sync/remotes/mod.rs
+++ b/asyncgit/src/sync/remotes/mod.rs
@@ -13,7 +13,7 @@ use crate::{
 	ProgressPercent,
 };
 use crossbeam_channel::Sender;
-use git2::{BranchType, FetchOptions, Repository};
+use git2::{BranchType, FetchOptions, ProxyOptions, Repository};
 use scopetime::scope_time;
 use utils::bytes2string;
 
@@ -24,6 +24,13 @@ use super::RepoPath;
 
 /// origin
 pub const DEFAULT_REMOTE_NAME: &str = "origin";
+
+///
+pub fn proxy_auto<'a>() -> ProxyOptions<'a> {
+	let mut proxy = ProxyOptions::new();
+	proxy.auto();
+	proxy
+}
 
 ///
 pub fn get_remotes(repo_path: &RepoPath) -> Result<Vec<String>> {
@@ -92,6 +99,7 @@ fn fetch_from_remote(
 	let mut options = FetchOptions::new();
 	let callbacks = Callbacks::new(progress_sender, basic_credential);
 	options.prune(git2::FetchPrune::On);
+	options.proxy_options(proxy_auto());
 	options.download_tags(git2::AutotagOption::All);
 	options.remote_callbacks(callbacks.callbacks());
 	remote.fetch(&[] as &[&str], Some(&mut options), None)?;
@@ -161,6 +169,7 @@ pub(crate) fn fetch(
 	options.download_tags(git2::AutotagOption::All);
 	let callbacks = Callbacks::new(progress_sender, basic_credential);
 	options.remote_callbacks(callbacks.callbacks());
+	options.proxy_options(proxy_auto());
 
 	remote.fetch(&[branch], Some(&mut options), None)?;
 

--- a/asyncgit/src/sync/remotes/push.rs
+++ b/asyncgit/src/sync/remotes/push.rs
@@ -2,8 +2,11 @@ use crate::{
 	error::{Error, Result},
 	progress::ProgressPercent,
 	sync::{
-		branch::branch_set_upstream, cred::BasicAuthCredential,
-		remotes::Callbacks, repository::repo, CommitId, RepoPath,
+		branch::branch_set_upstream,
+		cred::BasicAuthCredential,
+		remotes::{proxy_auto, Callbacks},
+		repository::repo,
+		CommitId, RepoPath,
 	},
 };
 use crossbeam_channel::Sender;
@@ -143,6 +146,7 @@ pub fn push_raw(
 	let mut remote = repo.find_remote(remote)?;
 
 	let mut options = PushOptions::new();
+	options.proxy_options(proxy_auto());
 
 	let callbacks = Callbacks::new(progress_sender, basic_credential);
 	options.remote_callbacks(callbacks.callbacks());

--- a/asyncgit/src/sync/remotes/tags.rs
+++ b/asyncgit/src/sync/remotes/tags.rs
@@ -5,8 +5,10 @@ use crate::{
 	error::Result,
 	progress::ProgressPercent,
 	sync::{
-		cred::BasicAuthCredential, remotes::Callbacks,
-		repository::repo, RepoPath,
+		cred::BasicAuthCredential,
+		remotes::{proxy_auto, Callbacks},
+		repository::repo,
+		RepoPath,
 	},
 };
 use crossbeam_channel::Sender;
@@ -59,7 +61,7 @@ fn remote_tag_refs(
 	let conn = remote.connect_auth(
 		Direction::Fetch,
 		Some(callbacks.callbacks()),
-		None,
+		Some(proxy_auto()),
 	)?;
 
 	let remote_heads = conn.list()?;
@@ -133,6 +135,7 @@ pub fn push_tags(
 			Callbacks::new(None, basic_credential.clone());
 		options.remote_callbacks(callbacks.callbacks());
 		options.packbuilder_parallelism(0);
+		options.proxy_options(proxy_auto());
 		remote.push(&[tag.as_str()], Some(&mut options))?;
 
 		progress_sender.as_ref().map(|sender| {

